test/parallel/test-abortcontroller.js

```diff
@@ -231,6 +233,30 @@ const { setTimeout: sleep } = require('timers/promises');
   AbortSignal.timeout(1_200_000);
 }
 
+{
+  (async () => {
+    // Test AbortSignal timeout can be GCed when no listeners exist
+    let ref;
+    {
+      function lis() {
+
+      }
+      ref = new globalThis.WeakRef(AbortSignal.timeout(1_200_000));
+      ref.deref().addEventListener('abort', lis);
+      aborted(ref.deref(), {});
+
+      await sleep(10);
+      globalThis.gc();
+
+      ref.deref().removeEventListener('abort', lis);
```

Comment:
maybe
```suggestion
      const signal = ref.deref();
      strictEqual(signal instanceof abortSignal, true);
      signal.removeEventListener('abort', lis);
```

Reply:
using deref and referencing this will fail the test as the reference is still used

Reply:
@rluvaton this const exists in a scope that ends prior to `gc`